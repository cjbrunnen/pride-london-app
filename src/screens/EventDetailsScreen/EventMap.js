// @flow
import React from "react";
import { TouchableOpacity, StyleSheet } from "react-native";
import MapView, { Marker } from "react-native-maps";
import showLocation from "./openMapLink";

type Props = {
  lat: number,
  lon: number,
  locationName: string
};

const EventMap = ({ lat, lon, locationName }: Props) => (
  <TouchableOpacity
    style={styles.mapWrapper}
    onPress={() => showLocation(lat, lon, locationName)}
  >
    <MapView
      style={styles.map}
      initialRegion={{
        latitude: lat,
        longitude: lon,
        latitudeDelta: 0.008,
        longitudeDelta: 0.008
      }}
      scrollEnabled={false}
      zoomEnabled={false}
      loadingEnabled
    >
      <Marker
        coordinate={{
          latitude: lat,
          longitude: lon
        }}
      />
    </MapView>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  mapWrapper: {
    borderRadius: 6,
    overflow: "hidden"
  },
  map: {
    height: 160
  }
});

export default EventMap;
