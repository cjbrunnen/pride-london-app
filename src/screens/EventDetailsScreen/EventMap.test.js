// @flow
import React from "react";
import { shallow } from "enzyme";
import EventMap from "./EventMap";

it("renders correctly", () => {
  const output = shallow(
    <EventMap lat={100} lon={200} locationName="somewhere" />
  );
  expect(output).toMatchSnapshot();
});
