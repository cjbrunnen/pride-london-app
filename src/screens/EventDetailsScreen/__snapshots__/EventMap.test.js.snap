// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "borderRadius": 6,
      "overflow": "hidden",
    }
  }
>
  <MapView
    initialRegion={
      Object {
        "latitude": 100,
        "latitudeDelta": 0.008,
        "longitude": 200,
        "longitudeDelta": 0.008,
      }
    }
    loadingEnabled={true}
    scrollEnabled={false}
    style={
      Object {
        "height": 160,
      }
    }
    zoomEnabled={false}
  >
    <MapMarker
      coordinate={
        Object {
          "latitude": 100,
          "longitude": 200,
        }
      }
      onPress={[Function]}
    />
  </MapView>
</TouchableOpacity>
`;
